src/map.rs

```diff
@@ -90,6 +96,103 @@ impl<T> Into<Option<T>> for RevOption<T> {
     }
 }
 
+pub enum Range<'a, 'b, K: 'a, V: 'a> where 'a: 'b {
+    /// Borrowed variant.
+    Borrowed(&'b [Option<(K, V)>], usize),
+    /// Owned variant, only available with the `std` or `alloc` feature enabled.
+    #[cfg(any(feature = "std", feature = "alloc"))]
+    Owned(BTreeRange<'b, K, V>),
+    _Phantom(&'a K),
```

Comment:
This should, at least, be `#[doc(hidden)]`, and I usually go with `__Phantom` (two underscores).